crates/nu-command/src/math/round.rs

```diff
@@ -80,7 +80,17 @@ impl Command for SubCommand {
 }
 
 fn operate(value: Value, head: Span, precision: Option<i64>) -> Value {
-    match value {
+    let mut result_value = value.clone();
```

Comment:
> **Warning**
> i did not run the full suite of tests, so this might not work 100% :wink: 

could we avoid this `clone` by applying the following patch?
```diff
diff --git a/crates/nu-command/src/math/round.rs b/crates/nu-command/src/math/round.rs
index 64b151221..1132c7681 100644
--- a/crates/nu-command/src/math/round.rs
+++ b/crates/nu-command/src/math/round.rs
@@ -80,7 +80,7 @@ impl Command for SubCommand {
 }
 
 fn operate(value: Value, head: Span, precision: Option<i64>) -> Value {
-    let mut result_value = value.clone();
+    let mut result_value = value;
 
     // We treat int values as float values in order to avoid code repetition in the match closure
     if let Value::Int { val, span } = result_value {
@@ -102,7 +102,6 @@ fn operate(value: Value, head: Span, precision: Option<i64>) -> Value {
                 span,
             },
         },
-        Value::Error { .. } => value,
         other => Value::Error {
             error: Box::new(ShellError::OnlySupportsThisInputType {
                 exp_input_type: "numeric".into(),
```

the `value` is only used in the `result_value` declaration and in the `Value::Error` arm and this does not change the behaviour of the `math round` command in the few examples i tried :open_mouth:

Reply:
we might even be able to simplify this further by applying
```diff
diff --git a/crates/nu-command/src/math/round.rs b/crates/nu-command/src/math/round.rs
index 64b151221..691dadd85 100644
--- a/crates/nu-command/src/math/round.rs
+++ b/crates/nu-command/src/math/round.rs
@@ -80,17 +80,17 @@ impl Command for SubCommand {
 }
 
 fn operate(value: Value, head: Span, precision: Option<i64>) -> Value {
-    let mut result_value = value.clone();
-
     // We treat int values as float values in order to avoid code repetition in the match closure
-    if let Value::Int { val, span } = result_value {
-        result_value = Value::Float {
+    let value = if let Value::Int { val, span } = value {
+        Value::Float {
             val: val as f64,
             span,
-        };
-    }
+        }
+    } else {
+        value
+    };
 
-    match result_value {
+    match value {
         Value::Float { val, span } => match precision {
             Some(precision_number) => Value::Float {
                 val: ((val * ((10_f64).powf(precision_number as f64))).round()
@@ -102,7 +102,6 @@ fn operate(value: Value, head: Span, precision: Option<i64>) -> Value {
                 span,
             },
         },
-        Value::Error { .. } => value,
         other => Value::Error {
             error: Box::new(ShellError::OnlySupportsThisInputType {
                 exp_input_type: "numeric".into(),
```

Reply:
> **Note**
> to apply these patches, you can just put the raw text in a `patch.diff` file and run
> `git apply patch.diff` :wink:

Reply:
Great tip!

Reply:
great thanks :star_struck: 
a perfect `git apply`cation, loving it :smirk: